src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/InstallTests.cs

```diff
@@ -0,0 +1,302 @@
+﻿using System;
+using NUnit.Framework;
+using Xamarin.ProjectTools;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using System.Text;
+using System.Xml.Linq;
+
+namespace Xamarin.Android.Build.Tests
+{
+	[TestFixture]
+	[NonParallelizable] //These tests deploy to devices
+	[Category ("Commercial")]
+	public class InstallTests : BaseTest
+	{
+		[Test]
+		public void ReInstallIfUserUninstalled ([Values (false, true)] bool isRelease)
+		{
+			if (!CommercialBuildAvailable)
+				Assert.Ignore ("Not required on Open Source Builds");
+
+			if (!HasDevices) {
+				Assert.Ignore ("Test Skipped no devices or emulators found.");
```

Comment:
monodroid has some `EmulatorHelper.csproj` that starts the emulator. Do we need something like that in this PR?

It might be we can use the _newer_ MSBuild tasks in this repo to do that.